lib/trackler/problem.rb

```diff
@@ -7,81 +7,108 @@ class Problem
     def initialize(slug, root)
       @slug = slug
       @root = root
+      @xcommon_repository_url = 'https://github.com/exercism/x-common/blob/master'
     end
 
     def exists?
-      !!md && !!yaml
+      description_exists? && yaml_exists?
     end
 
     def name
-      @name ||= slug.split('-').map(&:capitalize).join(' ')
+      @name ||= slug_to_name
     end
 
     def description
-      @description ||= File.read(filepath(md))
+      @description ||= load_description
     end
 
     def source_markdown
-      body = source.empty? ? "" : "%s" % source
-      url = source_url.empty? ? "" : "[%s](%s)" % [source_url, source_url]
-      if url.empty? && body.empty?
-        ""
-      else
-        "## Source\n\n" + [body, url].reject(&:empty?).join(" ")
-      end
+      text = [source, markdown_link(source_url)].reject(&:empty?).join(" ")
+      text.empty? ? text : "## Source\n\n#{text}"
     end
 
     def md_url
-      repo_url(md)
+      repo_url(md_path)
     end
 
     def json_url
-      repo_url(json) if !!json
+      repo_url(json_path) if json_path
     end
 
     def yaml_url
-      repo_url(yaml)
+      repo_url(yaml_path)
     end
 
-    %w(blurb source source_url).each do |name|
-      define_method name do
-        metadata[name].to_s.strip
-      end
+    def blurb
```

Comment:
It's possible to use the code coverage report to see if they're tested/used and possible to grep the code for 'def blurb' to discover where the method is defined.

If there were 100 I could see the benefit of the metaprogramming, when there are a handful just, inline them.